src/MongoDB/ServerApi.c

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright 2015-2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <php.h>
+#include <Zend/zend_interfaces.h>
+#include <ext/standard/php_var.h>
+#include <zend_smart_str.h>
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "phongo_compat.h"
+#include "php_phongo.h"
+
+zend_class_entry* php_phongo_serverapi_ce;
+
+static bool php_phongo_serverapi_create_libmongoc_object(mongoc_server_api_t **server_api, zend_string* version, bool strict_set, bool strict, bool deprecation_errors_set, bool deprecation_errors) /* {{{ */
+{
+	mongoc_server_api_version_t server_api_version;
+
+	if (!mongoc_server_api_version_from_string(ZSTR_VAL(version), &server_api_version)) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "Server API version \"%s\" is not supported in this driver version", ZSTR_VAL(version));
+		return false;
+	}
+
+	if (*server_api) {
+		mongoc_server_api_destroy(*server_api);
+	}
```

Comment:
Is this ever reached? If not, would it be better to throw an exception and return `false`? AFAICT, this is only called from the constructor and `php_phongo_serverapi_init_from_hash`, both of which should use freshly initialized objects.

Reply:
Replaced this with an exception instructing users to file a bug report as it shouldn't happen.

Reply:
Should we standardize on American English spelling ("initialized") here?

Reply:
I forgot we use Simplified English as opposed to Traditional English. I'll update it.